### PR TITLE
feat: added Pathways is search/All endpoint.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
feat: added LearnerPathways in search/all endpoint
* added LearnerPathways data is search/all endpoint
* added partner field is LearnerPathway model
* added missing tests for existing LearnerPathwaySerializer
* added tests related to newly added serializers and views

This data is hidden behind the include_learner_pathways feature flag.

example response: **GET http://localhost:18381/api/v1/search/all/?content_type=learnerpathway&include_learner_pathways=True**
```
{
    "count": 1,
    "next": null,
    "previous": null,
    "facets": {},
    "results": [
        {
            "aggregation_key": "learner_pathway:e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad",
            "content_type": "learnerpathway",
            "uuid": "e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad",
            "name": "Full stack developer",
            "status": "active",
            "banner_image": {
                "large": {
                    "url": "http://localhost:18381/media/media/learner_pathway/banner_images/e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad-93b03057f00e.large.png",
                    "width": 1440,
                    "height": 480
                },
                "medium": {
                    "url": "http://localhost:18381/media/media/learner_pathway/banner_images/e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad-93b03057f00e.medium.png",
                    "width": 726,
                    "height": 242
                },
                "small": {
                    "url": "http://localhost:18381/media/media/learner_pathway/banner_images/e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad-93b03057f00e.small.png",
                    "width": 435,
                    "height": 145
                },
                "x-small": {
                    "url": "http://localhost:18381/media/media/learner_pathway/banner_images/e246705d-9044-4bc9-8c8d-ebb0c3d0a9ad-93b03057f00e.x-small.png",
                    "width": 348,
                    "height": 116
                }
            },
            "overview": "This is a pathway to become a full stack developer.",
            "published": true,
            "skills": [
                {
                    "name": "My SKILL",
                    "description": "googe"
                }
            ],
            "skill_names": [
                "My SKILL"
            ],
            "partner": "edx",
            "steps": [
                {
                    "uuid": "63d708a7-8512-427e-8ae1-6ee8fa685360",
                    "min_requirement": 1,
                    "courses": [],
                    "programs": [
                        {
                            "uuid": "dcc9d1cf-a068-48c4-841d-934a0fcd2bfb",
                            "title": "edX Demonstration Program",
                            "short_description": "",
                            "card_image_url": "http://edx.devstack.lms:18000/asset-v1:edX+DemoX+Demo_Course+type@asset+block@images_course_image.jpg",
                            "content_type": "program"
                        }
                    ]
                },
                {
                    "uuid": "4a169c83-46f6-4a5a-8e58-5ccb76518f3d",
                    "min_requirement": 1,
                    "courses": [
                        {
                            "key": "edX+DemoX",
                            "title": "Demonstration Course",
                            "short_description": null,
                            "card_image_url": null,
                            "content_type": "course"
                        }
                    ],
                    "programs": []
                }
            ]
        }
    ]
}
```

### CODE REVIEW

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -59,30 +59,17 @@
...
-    # Convert a Django HTTPResponse object into a rest_framework.request
-    # using a generic API view. This is necessary because the drf-flex-fields
-    # library relies on the `.query_params` property of the request. DRF requests
-    # always have the `query_params` parameter unless the request is created using
-    # `APIRequestFactory`, which yelds Django's standard `HttpRequest`.
-    # Documentation: https://www.django-rest-framework.org/api-guide/testing/#forcing-authentication
-    # DRF issue: https://github.com/encode/django-rest-framework/issues/6488
-    return APIView().initialize_request(request)
-
-
```
Comment (HammadAhmadWaqas) [inline course_discovery/apps/api/tests/test_serializers.py:85]: make_request moved to test_utlity method to resolve circular import issue

File: course_discovery/apps/learner_pathway/tests/test_serializer.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from django.test import TestCase
+
+from course_discovery.apps.api.tests.test_utils import make_request
+from course_discovery.apps.learner_pathway.api.serializers import LearnerPathwaySerializer
+from course_discovery.apps.learner_pathway.tests.factories import (
+    LearnerPathwayCourseFactory, LearnerPathwayFactory, LearnerPathwayProgramFactory, LearnerPathwayStepFactory
+)
+
+
+class TestLearnerPathwaySerializer(TestCase):
```
Comment (HammadAhmadWaqas) [inline course_discovery/apps/learner_pathway/tests/test_serializer.py:10]: this test class is about an existing LearnerPathwaySerializer.

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    """
+
+    name = fields.TextField(
+        analyzer=synonym_text,
+        fields={
+            'suggest': fields.CompletionField(),
+            'edge_ngram_completion': fields.TextField(analyzer=edge_ngram_completion),
+        },
+    )
+    partner = fields.TextField(
```
Comment (irfanuddinahmad) [inline course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py:30]: Wouldn't a pathway have multiple partners?

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    )
+    status = fields.TextField()
+    overview = fields.TextField()
+    published = fields.BooleanField()
+    skill_names = fields.KeywordField(multi=True)
+    skills = fields.NestedField(properties={
+        'name': fields.TextField(),
+        'description': fields.TextField(),
+    })
+    steps = fields.KeywordField(multi=True)
```
Comment (irfanuddinahmad) [inline course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py]: Shouldn't the steps contain the keys and uuids of the component courses and programs?

File: course_discovery/apps/learner_pathway/migrations/0006_learnerpathway_partner.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('core', '0019_alter_user_first_name'),
+        ('learner_pathway', '0005_auto_20220120_2004'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='learnerpathway',
+            name='partner',
```
Comment (irfanuddinahmad) [inline course_discovery/apps/learner_pathway/migrations/0006_learnerpathway_partner.py:17]: see comment on the partner field above

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py (hunk 3 above)
Comment (HammadAhmadWaqas) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py:30]: All our existing models have one partner, for example, program and Course.

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py (hunk 4 above)
Comment (HammadAhmadWaqas) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py]: These keys are just being indexed in elastic search. Other data serialize in the serializers.

File: course_discovery/apps/learner_pathway/migrations/0006_learnerpathway_partner.py (hunk 5 above)
Comment (HammadAhmadWaqas) [reply course_discovery/apps/learner_pathway/migrations/0006_learnerpathway_partner.py:17]: See answer on partner field

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py (hunk 3 above)
Comment (irfanuddinahmad) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py:30]: Can we ping Nathan to double check?

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py (hunk 4 above)
Comment (irfanuddinahmad) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py]: Why are we indexing these uuids fields?

Comment (HammadAhmadWaqas) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py]: We are following the course document [here](https://github.com/openedx/course-discovery/blob/ad1af6e122fc19250f85387d5d4447c48b42ff3e/course_discovery/apps/course_metadata/search_indexes/documents/course.py#L60). This PathwayDoucment can be used in a serializer where we don't want to get step details as well. For now, if we want we can remove it from the index.

File: course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py (hunk 3 above)
Comment (HammadAhmadWaqas) [reply course_discovery/apps/course_metadata/search_indexes/documents/learner_pathway.py:30]: Sure.

**Review by saleem-latif — APPROVED** [review]
Looks Good  👍

**Review by irfanuddinahmad — APPROVED** [review]
LGTM! ... rebase with master

**Pull request merged**